From 75ba7fdf8eb25198e7ab483433c0a662475d49bc Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 4 Apr 2019 11:07:43 +0800
Subject: [PATCH 01/30] ARM64: DMA: fix compile error for
 swiotlb_dma_ops.dma_supported

In upstream commit b7da40949ec5 ("arm64: rename swiotlb_dma_ops"),
swiotlb_dma_ops in dma-mapping.c has been renamed to arm64_swiotlb_dma_ops,
so __swiotlb_dma_supported_quirk should be assigned to
arm64_swiotlb_dma_ops.dma_supported but not swiotlb_dma_ops.dma_supported.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/arm64/mm/dma-mapping.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/mm/dma-mapping.c b/arch/arm64/mm/dma-mapping.c
index 629446866951..31579aa69aeb 100644
--- a/arch/arm64/mm/dma-mapping.c
+++ b/arch/arm64/mm/dma-mapping.c
@@ -930,7 +930,7 @@ void arch_setup_dma_ops(struct device *dev, u64 dma_base, u64 size,
 	if (of_property_read_u32(np, "only-dma-mask32", &mask32))
 		mask32 = 0;
 	if (mask32) {
-		swiotlb_dma_ops.dma_supported = __swiotlb_dma_supported_quirk;
+		arm64_swiotlb_dma_ops.dma_supported = __swiotlb_dma_supported_quirk;
 #ifdef CONFIG_IOMMU_DMA
 		iommu_dma_ops.dma_supported = iommu_dma_supported_quirk;
 #endif
-- 
2.17.1

